apps/manager_app/auth_card.c

```diff
@@ -605,47 +525,24 @@ void card_auth_handler(manager_query_t *query) {
               .message = "",
               .pair_card_required = false,
           },
-      .query = query};
-
-  core_status_set_flow_status(MANAGER_AUTH_CARD_STATUS_INIT);
-  do {
-    /* TODO: Create a handler to resolve specific returns and either send error
-     * response or abort flow accordingly*/
-    if (MANAGER_TASK_SUCCESS !=
-        handle_auth_card_query(&auth_card_data, &resp)) {
-      usb_clear_event();
-      return;
-    }
+      .query = query,
+      .flow_status = MANAGER_AUTH_CARD_STATUS_INIT};
 
-    send_auth_card_response(&resp);
-    /**
-     * TODO: Sending errors and response at this point is not ideal as host gets
-     * indicated of the issue after the error dispaly completes
-     * - Each handler should send their own response to host for success case
-     * - Send error response to host in errors cases before showing error
-     * display.
-     * - The parent should only be concerned with correct exit condition i.e.
-     * flow complete or error occurred
-     * - Flow complete should be indicated by flow status.
-     * - For early exits, say auth failed after the serial sign, handler can
-     * return MANAGER_TASK_FAILED to exit.
-     */
-    if (MANAGER_AUTH_CARD_RESPONSE_COMMON_ERROR_TAG == resp.which_response) {
+  while (1) {
+    if (MANAGER_TASK_SUCCESS != handle_auth_card_query(&auth_card_data)) {
+      usb_clear_event();
       return;
     }
-    memzero(&resp, sizeof(resp));
 
-    if (FLOW_COMPLETE_STATE) {
+    if (FLOW_COMPLETE_STATE(&auth_card_data)) {
       return;
     }
 
-    if (MANAGER_TASK_SUCCESS !=
-        get_and_decode_valid_query_from_host(auth_card_data.query)) {
+    if (false ==
+        manager_get_query(auth_card_data.query, MANAGER_QUERY_AUTH_CARD_TAG)) {
       usb_clear_event();
```

Comment:
No need for this, an appropriate response is sent from within manager_get_query

Reply:
updated